pkg/planner/planner.go

```diff
@@ -415,33 +415,26 @@ func (p *Planner) desiredPlan(cluster *rkev1.RKECluster, secret plan.Secret, ent
 
 	result.Files = append(result.Files, plan.File{
 		Content: base64.StdEncoding.EncodeToString(configData),
-		Path:    fmt.Sprintf("/etc/rancher/%s/config.yaml", GetRuntime(cluster)),
+		Path:    fmt.Sprintf("/etc/rancher/%s/config.yaml", GetRuntime(cluster.Spec.KubernetesVersion)),
 	})
 
 	return result, nil
 }
 
-func GetRuntime(cluster *rkev1.RKECluster) string {
-	if strings.Contains(cluster.Spec.KubernetesVersion, RuntimeK3S) {
+func GetRuntime(kubernetesVersion string) string {
```

Comment:
Is there a reason this was moved back to being just the version?  The reason that this was moved to be whole cluster object is just that the current check was viewed as a short term hack to test k3s, but in the future the runtime will be determined by the existence of `k3sConfig` field.  So I didn't want too much code to just pass the version cause it could be in the future when we make that change the caller won't have the full cluster object making the refactor harder.

Reply:
Understood. I changed it so that I could use it with the `rancher.cattle.io` cluster as well in `pkg/controllers/rke/machine/controller.go`.

I need the rancher cluster here to get the kubeconfig with the `kubeconfigManager`.

Reply:
This will probably have to change again once the types are settled.